app/models/user.rb

```diff
@@ -26,4 +26,14 @@ def follows?(another_user)
   def can_follow?(another_user)
     !(self.follows?(another_user) || self == another_user)
   end
+
+  def update_with_token!
+    update_column(:token, generate_token)
+  end
+
+  private
+
+  def generate_token
+    self.token = SecureRandom.urlsafe_base64
```

Comment:
I am not sure why you set the token here. Why `self.token = SecureRandom.urlsafe_base64` ? You don't need that attribute set in this method - thats what you `update_with_token!` does. The goal of this method is to do ONLY to generate the token and return it.